### PR TITLE
Make it easier to add new language support

### PR DESCRIPTION
**Pull request by shivamdev-lgtm**
This still needs a lot of work, but is a good start

### CODE REVIEW

**Comment by Rexogamer** [comment]
please note the changes on my fork to how the editor selection works

**Comment by shivamdev-lgtm** [comment]
> please note the changes on my fork to how the editor selection works

This commit is quite unrelated to IDE selection.

**Comment by Rexogamer** [comment]
Fair enough, I was just making you aware.

File: automate-project.py [hunk 1]
```diff
@@ -91,7 +85,7 @@ def CreateGitHubRepo():
     GetCredentials()
     try:
         user = Github(username, password).get_user()
-        user.create_repo(repoName)
+        # user.create_repo(repoName)
```
Comment (Rexogamer) [inline automate-project.py]: Are you sure this should be a comment?

Comment (shivamdev-lgtm) [reply automate-project.py]: Oh i forgot to uncomment that after debugging. Will fix.

**Comment by Rexogamer** [comment]
I disagree with the changes to script.config (making it an example you must copy), and feel we should stick to just having the user edit the file. The base config works fine on Windows.

**Comment by shivamdev-lgtm** [comment]
@Rexogamer I feel it's not a good idea to track a file via git which may contain sensitive informations like username and password and might also reveal information about your environment. One might accidentally add their username and passwords and commit it with other changes and then push it to GitHub making their credentials public. It's just how most of the projects work on GitHub. They never track such sensitive files

**Comment by Rexogamer** [comment]
By default the script.config was empty on that front. I get what you mean but I dunno if the average user would?

**Comment by jarodburchill** [comment]
I understand the logic behind trying to separate the concerns of the project-type processes, however, turning them into JSON data does not seems like the way to go to me… Having to convert the JSON files just seems a lot more unreadable and unnecessary. Also we will end up with a whole bunch of really short JSON files in the languages dir. What if we just separated the project-type processes into their own python module?

**Review by jarodburchill — COMMENTED** [review]
I understand the logic behind trying to separate the concerns of the project-type processes, however, turning them into JSON data does not seems like the way to go to me… Having to convert the JSON files just seems a lot more unreadable and unnecessary. Also we will end up with a whole bunch of really short JSON files in the languages dir. What if we just separated the project-type processes into their own python module?

**Comment by shivamdev-lgtm** [comment]
@jarodburchill well I'm quite neutral about both JSON and separating it into python modules. The reason I saw JSON more promising was any average user can add support for their whatever language they are working on. A python module on the other hand might give us more flexibility(which we can still get with JSON if we work on it) but that would take the ability to quickly add support for any language.

**Comment by jarodburchill** [comment]
> @Rexogamer I feel it's not a good idea to track a file via git which may contain sensitive informations like username and password and might also reveal information about your environment. One might accidentally add their username and passwords and commit it with other changes and then push it to GitHub making their credentials public. It's just how most of the projects work on GitHub. They never track such sensitive files

On this front, I see your point, but the username and password fields in the config are not required. You can enter them manually each time you run it.

**Comment by shivamdev-lgtm** [comment]
> > @Rexogamer I feel it's not a good idea to track a file via git which may contain sensitive informations like username and password and might also reveal information about your environment. One might accidentally add their username and passwords and commit it with other changes and then push it to GitHub making their credentials public. It's just how most of the projects work on GitHub. They never track such sensitive files
> 
> On this front, I see your point, but the username and password fields in the config are not required. You can enter them manually each time you run it.

Yes, indeed but the field is there to be used, people (including me), don't like to enter username and password every time. I just faced the following problem:
I like to keep my file at C:\dev, so I edited the config file which got tracked by git. So I have to edit the file every time I have to push a commit to github. Isn't it more easier to set it up once then to edit the file every time one wants to push a change?

**Comment by Rexogamer** [comment]
> > > @Rexogamer I feel it's not a good idea to track a file via git which may contain sensitive informations like username and password and might also reveal information about your environment. One might accidentally add their username and passwords and commit it with other changes and then push it to GitHub making their credentials public. It's just how most of the projects work on GitHub. They never track such sensitive files
> > 
> > 
> > On this front, I see your point, but the username and password fields in the config are not required. You can enter them manually each time you run it.
> 
> Yes, indeed but the field is there to be used, people (including me), don't like to enter username and password every time. I just faced the following problem:
> I like to keep my file at C:\dev, so I edited the config file which got tracked by git. So I have to edit the file every time I have to push a commit to github. Isn't it more easier to set it up once then to edit the file every time one wants to push a change?

Well... why not keep script.config as it was, but use a gitignore?

**Comment by jarodburchill** [comment]
I like some of the ideas you have here, but I think this pull request is not the direction I want to go. I will definitely find a way to have better language support and a more secure way of storing passwords, but this approach would not be the way I would implement it. Thanks for your effort and ideas! Feel free to submit another pull request if you have more ideas or want to follow on the same ideas in a different way

**Pull request closed without merging**